python/ray/tune/tune.py

```diff
@@ -264,14 +267,119 @@ def run(
             ``ray.tune.callback.Callback`` class. If not passed,
             `LoggerCallback` and `SyncerCallback` callbacks are automatically
             added.
-
+        _remote (bool): Whether to run the Tune driver in a remote function.
+            This is disabled automatically if a custom trial executor is
+            passed in. This is enabled by default in Ray client mode.
 
     Returns:
         ExperimentAnalysis: Object for experiment analysis.
 
     Raises:
         TuneError: Any trials failed and `raise_on_failed_trial` is True.
     """
+
+    if _remote is None:
+        _remote = ray.util.client.ray.is_connected()
+    if _remote and not trial_executor:
+        _ray_auto_init()
+        get_fn = ray.get
+        exec_impl = ray.remote(num_cpus=0)(_tune_run_impl).remote
+    else:
+        get_fn = lambda x: x  # noqa
```

Comment:
Maybe throw a warning/exception if `remote=True` and a trial executor have been passed explicitly

Reply:
(feel free to ignore though since this is a private parameter for now)